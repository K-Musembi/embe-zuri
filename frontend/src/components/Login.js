import React, { useState } from "react";
import { API_URL } from "../config";

const Login = ({ onLogin }) => {
    const [username, setUsername] = useState("");
    const [password, setPassword] = useState("");

    const handleLogin = async (event) => {
        event.preventDefault();

        const response = await fetch(`${API_URL}/login`, {
            method: "POST",
            headers: { "Content-Type": "application/json" },
            body: JSON.stringify({ username, password }),
        });
        if (response.ok) {
            const result = await response.json();
            onLogin(result.token);
            alert("Successful login!")
        } else {
            alert("Check your credentials and try again.")
        }
    };

    return (
        <div className="login-container">
            <h2>Login</h2>
            <form onSubmit={handleLogin}>
                <label>
                    Username
                    <input type="text" value={username} onChange={(event) => setUsername(event.target.value)} required />
                </label>
                <label>
                    password
                    <input type="password" value={password} onChange={(event) => setPassword(event.target.value)} required />
                </label>
                <button type="submit">Submit</button>
            </form>
        </div>
    );
};

export default Login;
