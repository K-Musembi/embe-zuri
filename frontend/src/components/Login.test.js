import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Login from './Login';

// Mock fetch globally
global.fetch = jest.fn(() =>
  Promise.resolve({
    ok: true,
    json: () => Promise.resolve({ token: 'fake-token' })
  })
);

describe('Login Component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <Login onLogin={() => {}} />
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
