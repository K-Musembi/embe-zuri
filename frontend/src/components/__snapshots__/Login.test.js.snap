// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component renders correctly 1`] = `
<div
  className="signup-login-container"
>
  <h2>
    Login
  </h2>
  <form
    onSubmit={[Function]}
  >
    <label>
      Username
      <input
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </label>
    <label>
      password
      <input
        onChange={[Function]}
        required={true}
        type="password"
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Submit
    </button>
    <p>
      Not yet registered? 
      <a
        href="/signup"
      >
        Sign Up
      </a>
    </p>
  </form>
</div>
`;
